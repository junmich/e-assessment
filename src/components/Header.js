import React from 'react';

const Header = ({ title }) => {
    return (
        <div className="header-title">{title}</div>
    )
};

Header.defaultProps = {
    title: 'Demo Product App'
};

export default Header;