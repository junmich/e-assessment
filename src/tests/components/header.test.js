import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

describe('Header Test', () => {
    test('should render header correctly', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('.header-title').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });
    test('should render header with title', () => {
        const title = "My App";
        const wrapper = shallow(<Header title={title}/>);
        expect(wrapper.find('.header-title').length).toBe(1);
        expect(wrapper.find('.header-title').text()).toBe(title);
        expect(wrapper).toMatchSnapshot();
    });
});

