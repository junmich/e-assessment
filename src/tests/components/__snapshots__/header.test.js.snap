// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test should render header correctly 1`] = `
<div
  className="header-title"
>
  Demo Product App
</div>
`;

exports[`Header Test should render header with title 1`] = `
<div
  className="header-title"
>
  My App
</div>
`;
